sqlitedict.py

```diff
@@ -98,7 +98,13 @@ def encode(obj):
 
 def decode(obj):
     """Deserialize objects retrieved from SQLite."""
-    return loads(bytes(obj))
+    try:
+        return loads(bytes(obj))
+    except (UnpicklingError, ValueError, EOFError, TypeError):  # Checking for the possible DB used by version 1.2.
+        logger.warning("Not a pickled string %s in sqlite, probably compatibility with sqlite 2.0 problem. "
+            "you can simply run `python sqlite_migration.py` ... in order to migrate the sqlite DB", obj)
+        raise KeyError("your sqlite db is not compatibile with sqlitedict2.0 "
+            "you can simply run `python sqlite_migration.py` ... in order to migrate the sqlite DB")
```

Comment:
You should probably avoid throwing away the original exception. There are many different cases where such exceptions may be raised but are legitimate, such as if you pickle a class instance of a different signature than the definition given by the same module during unpickling.